packages/api/tests/api/project/contributors/put.test.ts

```diff
@@ -0,0 +1,115 @@
+  });
+
+  it('should return 409 if the user is already a contributor', async () => {
+    validatePayloadMock.mockReturnValueOnce({ errorHandled: false, data: validPayload } as any);
+    const req = createMockRequest({ user: mockUser, ...validPayload } as any);
+    const res = createMockResponse();
+    const mockProject = { inviteCode: validPayload.inviteCode, contributors: { add: jest.fn() } };
+    req.entityManager.findOneOrFail.mockResolvedValueOnce(mockProject);
+    (req.entityManager.transactional as jest.Mock).mockRejectedValueOnce({ name: 'NotFoundError' });
+
+    await put(req as any, res as any);
+
+    expect(req.entityManager.findOneOrFail).toBeCalledWith(Project, { id: validPayload.projectId });
+    expect(mockProject.contributors.add).not.toBeCalled();
+    expect(req.entityManager.persist).not.toBeCalled();
+    expect(res.sendStatus).toHaveBeenCalledWith(409);
+  });
+
+  it('should return 423 if the row lock could not be obtained', async () => {
+    validatePayloadMock.mockReturnValueOnce({ errorHandled: false, data: validPayload } as any);
+    const req = createMockRequest({ user: mockUser, ...validPayload } as any);
+    const res = createMockResponse();
+    const mockProject = { inviteCode: validPayload.inviteCode, contributors: { add: jest.fn() } };
+    req.entityManager.findOneOrFail.mockResolvedValueOnce(mockProject);
+    req.entityManager.transactional.mockRejectedValueOnce({ code: '55P03' });
+
+    await put(req as any, res as any);
+
+    expect(req.entityManager.findOneOrFail).toBeCalledWith(Project, { id: validPayload.projectId });
+    expect(mockProject.contributors.add).not.toBeCalled();
+    expect(req.entityManager.persist).not.toBeCalled();
+    expect(res.sendStatus).toHaveBeenCalledWith(423);
+  });
+
+  it('should return 500 something else goes wrong', async () => {
+    validatePayloadMock.mockReturnValueOnce({ errorHandled: false, data: validPayload } as any);
+    const req = createMockRequest({ user: mockUser, ...validPayload } as any);
+    const res = createMockResponse();
+    const mockProject = { inviteCode: validPayload.inviteCode, contributors: { add: jest.fn() } };
+    req.entityManager.findOneOrFail.mockResolvedValueOnce(mockProject);
+    req.entityManager.transactional.mockRejectedValueOnce(new Error('Oh no!'));
+
+    await put(req as any, res as any);
+
+    expect(req.entityManager.findOneOrFail).toBeCalledWith(Project, { id: validPayload.projectId });
+    expect(mockProject.contributors.add).not.toBeCalled();
+    expect(req.entityManager.persist).not.toBeCalled();
+    expect(res.sendStatus).toHaveBeenCalledWith(500);
+  });
+
+  it('bails if a validation error occurs', async () => {
+    validatePayloadMock.mockReturnValueOnce({ errorHandled: true });
+    const req = createMockRequest();
+    const res = createMockResponse();
+
+    await put(req as any, res as any);
+
+    expect(validatePayloadMock).toBeCalledTimes(1);
```

Comment:
Can you add another expect below this to make sure `transactional` was not called?